mewarpx/mewarpx/runinfo.py

```diff
@@ -0,0 +1,397 @@
+"""Methods for storing parameters from a run.
+
+This should all be made to Pickle easily, so that we can load the parameters at
+a later time.
+"""
+import collections
+import copy
+import datetime
+import logging
+import os
+import platform
+import subprocess
+
+import dill
+import mewarpx
+import numpy as np
+
+import mewarpx.util as mwxutil
+
+from mewarpx.mwxrun import mwxrun
+from functools import reduce
+
+logger = logging.getLogger(__name__)
+
+class SimInfo(object):
+
+    """Store basic simulation parameters used throughout analysis.
+
+    This must contain:
+        SimInfo.nxyz (int nx, int ny, int nz)
+        SimInfo.pos_lims (floats xmin, xmax, ymin, ymax, zmin, zmax)
+        SimInfo.geom (str 'XZ', 'RZ' or 'XYZ')
+        SimInfo.dt (float)
+        SimInfo.periodic (bool)
+
+    Arguments:
+        nxyz (int, int, int): tuple of the grid dimensions
+        pos_lims (floats xmin, xmax, ymin, ymax, zmin, zmax): tuple of the
+            position limits along each dimension
```

Comment:
I think for sphinx the types should be tuple, and then you could specify the types inside the tuple in the description of the parameter. ie.
```suggestion
        nxyz (tuple): tuple of ints, holding the grid dimensions
        pos_lims (tuple) : tuple of floats, holding the position limits along each 
            dimension (xmax, ymin, ymax, zmin, zmax)
```